realtime/server.js

```diff
@@ -70,25 +77,26 @@ socket.on('connection', function (client) {
             sendLoginErrorResponse(username, client, 'chars');
             return;
         }
-        if (usernames[username]) {
+        if (usernames[username]) { 
             sendLoginErrorResponse(username, client, 'taken');
             return;
         }
-        people[client.id] = username;
-        usernames[username] = true;
+        people[client.id] = username; //people has value 'username' at position 'client.id'
+        usernames[username] = true; //set the current name as true, meaning that exists for now.
         resp.people = people;
         winston.info('[' + username + '] login');
         logUsersCount();
         client.emit('login-response', resp);
         socket.sockets.emit('join', username);
     });
-
+    
+    //when user sends a message, this function is executed
     client.on('message', function(data) {
-        var text = data.text;
-        data.username = people[client.id];
-        socket.sockets.emit('message', data);
+        var text = data.text; //text of the message
+        data.username = people[client.id]; //data.username keeps the username value
+        socket.sockets.emit('message', data); //send back to the client that we received the message
```

Comment:
This comment is wrong. We are not sending it to the particular client, but instead we are broadcasting it.